### DISCUSSION TITLE
Could I use chro:positon instead of rsID for analyze?

### DISCUSSION THREAD

**Discussion by sjidong12** [MRCIEU/TwoSampleMR#372] [Q&A]
I have a data which is using chro:position instead of rsID in the GWAS summary data. So could I use it as outcome in the MR study?

If not, how can I transfer it into rsID the analyze requested?

Thank you!

**Comment by rchapleau** [comment]
@sjidong12 

I found a work-around to this issue using the ieugwasr and dplyr packages. I was able to do it in one line, but I'll walk you through the steps in case somebody else runs into this issue. My input data was from the NIH's dbGaP. Your column names may be different.

Step 1) Create a column in the dataframe with chr:pos
**df$chrpos <- paste(df$CHR, df$BP, sep=":")**

Step 2) Create a list of rsIDs from the summary statistics file
**rsID_list <- variants_to_rsid(df$chrpos)**

Step 3) Create a dataframe of the variant information
**rsID <- variants_rsid(rsID_list)**

Step 4) Create a new dataframe with the original GWAS results and the new rsIDs, missing rsIDs are "NA" (The rsID dataframe should have the 'chr' and 'pos' column names while my original had 'CHR' and 'BP')
**df2 <- merge(rsID, df, by.x=c('chr', 'pos'), by.y=c('CHR', 'BP'), all=TRUE)**

Step 5) Create a final dataframe that contains all of the original information and all of the new rsIDs. Some loci may have multiple SNPs, so creating a single column in the original file may result in an "Error in `$<-.data.frame`(`*tmp*`, rsid, value = list(query = c(NA, NA,  :  replacement has 168966 rows, data has 168960" (the row counts may be different for you)
**df3 <- merge(df2, df, all=TRUE)**

You can do everything in a single line and restrict the new dataframe to only those columns necessary for running TwoSampleMR (your column indices at the end may be different).

"One liner"
**df2 <- merge(merge(variants_rsid(variants_to_rsid(paste(df$CHR, df$BP, sep=":"))), df, by.x=c('chr', 'pos'), by.y=c('CHR', 'BP'), all=TRUE), df, all=TRUE)[, c(1,2,3,5,6,7,8,9,10,12,13,14)]**

This final one liner ends up only creating a single dataframe. In my case, I had an extra 6 rows added because of duplicate variant names.

My original dataframe:
CHR	snpID	BP	effect_allele	eaf	F_U	other_allele	CHISQ	pval	OR	beta	se	chrpos
1	SNP_A-2236359	782343	C	0.04765	0.02706	T	11.03	0.0008965	1.799	0.587231	0.176816	1:782343

My final dataframe:
snpID	effect_allele	eaf	other_allele	CHISQ	pval	OR	beta	se	chr	pos	rsid
SNP_A-8459285	C	0.2884	A	13.88	0.0001948	1.331	0.2859305	0.07674615	4	24516008	rs10026414

I hope this helps.
